recipes/systemc-cci/1.0.0/conanfile.py

```diff
@@ -0,0 +1,70 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class SystemccciConan(ConanFile):
+    name = "systemc-cci"
+    version = "1.0.0"
+    description = """SystemC Configuration, Control and Inspection library"""
+    homepage = "https://www.accellera.org/"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"
+    topics = ("simulation", "modeling", "esl", "cci")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True
+    }
+    requires = "systemc/2.3.3"
+    generators = "make"
+    exports_sources = "patches/**"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("Windows build not supported")
+
+        tools.check_min_cppstd(self, "11")
```

Comment:
```suggestion
	    if self.settings.get_safe("compiler.cppstd"):
            tools.check_min_cppstd(self, "11")
```

Reply:
Why is this change needed, as far as I can see `tools.check_min_cppstd` already takes care of this?

https://github.com/conan-io/conan/blob/a1086d3df17b482a8ab6e0ce506251dfc7327381/conans/client/tools/settings.py#L41-L43

Reply:
No. If cppstd is not available (and it's not available by default), it will check the default c++ standard version supported by the compiler. Since GCC 4.7 you can use C++11 (not all features), but it doesn't use C++11 by default, so it will break.

Reply:
Ok but that is exactly the intention, either the compiler supports C++11 by default or it is told to use C++11. 

If C++98 would be used the code will not build so skipping the check does not look like the correct approach.

Reply:
No. again, you are striping a compiler version which would work (e.g. gcc5). If the makefile pass -std=c++11, Conan doesn't care, it won't check it, it will check if the compiler runs c++11 by default, if not, it will raise an ConanInvalidConfiguration.

Checking compiler.cppstd it's very regular approach, you will find in 99% of recipes in CCI which require >=C++11.

Reply:
Ok I doubled  checked and the code in the PR is exactly the behaviour as I intended.

The first point to note is that the `SystemC CCI` Makefile does not contain `-std=c++11` so if the proposed `if` check is put in place the build will fail if the default C++ standard for a compiler is C++ 98 (eg. gcc version < 6).

Further the `SystemC` dependency has the C++ standard version embedded in one of its namespaces, which is declared in a public header file and defined in a source file. So if you would force `-std=c++11`in the `SystemC CCI` Makefile it will fail to link against the `SystemC` dependency if the compilers default c++ standard is not C++11.

So the check as proposed in the PR looks correct to me:
1) either the compiler supports c++11 or greater by default (gcc >= 6.0.0)
2) the user sets compiler.cppstd to 11 or greater

Reply:
@vanwinkeljan `compiler.cppstd` being `None` can mean 2 things: 1. the user uses the default c++ standard or 2. he overrides it and does not bother set the conan standard.

I interpret `cppstd=None` as uninitialized. The actual standard can be everything.
So the recipe should do its best to have a package available.

Reply:
Hmm I did interpret `cppstd=None` as use default compiler version.

And I get the same feeling from the documentation but this could be a wrong interpretation of the documentation: https://docs.conan.io/en/1.7/howtos/manage_cpp_standard.html

Anway what is the correct way forward here?
One option would be to set the ccpstd to 11 for both `SystemC CCI` and `SystemC` in case no cppstd is given.

Reply:
The correct way is the change that I suggested you above. 

> One option would be to set the ccpstd to 11

No. You can't force any setting in a recipe, it's a prone error.

Reply:
> The correct way is the change that I suggested you above.

@uilianries if the change is applied CI will fail due to the reason mentioned in https://github.com/conan-io/conan-center-index/pull/1643#discussion_r431401116

Reply:
Reference to the code to prevent to apply proposed change:
https://github.com/accellera-official/systemc/blob/604182509559ae42c34e878f508bae7c18abbbd6/src/sysc/kernel/sc_ver.h#L89-L106

Reply:
I forgot that these systemc embed the c++ version in their namespace.
So it should use the same cppstd as the systemc package (which works using c++98).
If that's guaranteed, then a raise on an default cppstd is indeed justified imho.

Reply:
@vanwinkeljan you don't an error because `check_min_cppstd` raises `ConanInvalidConfiguration` which is skipped by CCI. You will not find GCC 4.9 and 5 listed in the builds successfully.

Reply:
@vanwinkeljan if systemc-cci requires c++11, why -std=c++11 is not declared in the Makefile?

Reply:
> if systemc-cci requires c++11, why -std=c++11 is not declared in the Makefile?

@uilianries actually it is a bit more complicated the required version is the same version as the version used to build the SystemC library with a minimum version of C++ 11 (due to the syntax used in CCI)

> you don't see an error because check_min_cppstd raises ConanInvalidConfiguration which is skipped by CCI. You will not find GCC 4.9 and 5 listed in the builds successfully.

Fyi I did test the proposed change locally with a compiler that uses C++98 as default.
In case the Makefile is not updated the code fails to compile, if `-std=c++11` is added to the Makefile the test application will fail to link as the `SystemC CCI` and `SytemC` library are not build with the same C++ standard.

Reply:
@uilianries any further comments on the feedback above?

Reply:
This recipe it's a very edge-case. It requires c++11, but it doesn't have any flag about, but also need to be aligned to systemC, including its c++ version. In this case, it's fine accepting it, but only because this particular case. For any other regular case, it's not acceptable, because it excludes some compiler versions from the build line.

Reply:
@uilianries Shall I add a comment in the code to highlight this?

Reply:
@vanwinkeljan not required, because you are not forcing an option or setting.